docs/dev/testing.md

```diff
@@ -0,0 +1,62 @@
+# Assisted Installer Testing
+
+The assisted-installer tests are divided into 3 categories:
+
+* **Unit tests** - Focused on a module/function level while other modules are mocked.
+Unit tests are located next to a module named `module_name_test.go`
+
+* **Subsystem tests** - Focused on the component while mocking other component. (e.g. agent/installer results).
+
+* **System tests** (a.k.a e2e) - Running full flows with all components.
+The e2e tests are divided into u/s (upstream) basic workflows on [assisted-test-infra](https://github.com/openshift/assisted-test-infra/tree/master/discovery-infra/tests) and d/s (downstream) extended regression tests maintained by both DEV and QE teams on [kni-assisted-installer-auto](https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/tree/master/api_tests).
+
+## Repository CI
+
+Assisted-service CI jobs are defined under [openshift/release](https://github.com/openshift/release) on [openshift-assisted-service-master.yaml](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml).
+
+All the available jobs can be viewed on [Openshift CI Step Registry](https://steps.ci.openshift.org/search?job=openshift-assisted-service).
+
+Read more about OpenShift CI infrastructure on [OpenShift CI Docs](https://docs.ci.openshift.org/docs/).
+
+### Adding a new CI job
+
+When adding a new job the following rules of thumbs should be taken into account:
+
+1. Test logic needs to be maintained in the repository under test and not under openshift/release.
+It would allow easier integration with other tools, less dependency of the CI infrastructure, and most importantly the availability to run it locally.
+
+1. When introducing a new job it should be both a presubmit and a periodic. A presubmit needs to be available so contributors would be able to run it whenever they want.
+The presubmit needs to be configured as `always_run: false` and `optional: true` (not blocking a merge) until proving stability.
+The periodic needs to run on a frequent basis (e.g. daily) and have a `reporter_config` configured, in order to be notified on Slack whenever there's a breakage.
+
+1. In case the new job affects multiple repositories - every repository should have the same presubmit job so it could be tested for every component change. For example [e2e-metal-assisted-olm](https://steps.ci.openshift.org/search?job=e2e-metal-assisted-olm).
+
+[An example of a PR adding a new job](https://github.com/openshift/release/pull/21604)
+
+### FAQ
+
+#### **How can I debug CI failures?**
+
+A CI job can be debugged only in runtime.
+Once a job terminates it can no longer be debugged.
+In order to .
+Follow [Debug Prow Jobs live](https://docs.google.com/document/d/1mxLQskpLI7b6Ndw-TW6i_aXQqK1DGNgFc4tP_4jt8tU) document.
```

Comment:
This is a private document I believe. Also to debug them live you need to be a part of the Openshift org here in Github. We should probably mention this 🤔

Reply:
Adding it to Openshift CI docs is a bit more work, and also the right thing to do

We can alternatively just suggest Omer's automatic script instead of this. It's a github gist iirc

Reply:
Okay, let's start with that.

Reply:
Openshift CI docs has the following page:
https://docs.ci.openshift.org/docs/how-tos/interact-with-running-jobs/